docs: attempt to get us back in alignment with RFC-2119 (#133)
conventional-commits/conventionalcommits.org: content/next/index.md

---
draft: true

> aliases: ["/en/"]

Reviewer: This one should have the alias 😄

Author: I'm not quite sure what this value should be? I was copying the old document in this folder.

---

> # Conventional Commits 1.0.0-beta.4

Reviewer: 👍

fixes issue #12
```

## Specification

The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).

> 1. Commits MUST be prefixed with a type, which consists of a noun, `feat`, `fix`, etc., followed

Reviewer: 👍

  by an OPTIONAL scope, and a REQUIRED terminal semicolon and space.
1. The type `feat` MUST be used when a commit adds a new feature to your application or library.
1. The type `fix` MUST be used when a commit represents a bug fix for your application.

> 1. An optional scope MAY be provided after a type. A scope MUST consist of a noun describing a

Reviewer 2: `An optional scope` => `A scope`

Since `optional` and `MAY` feel redundant.

Author: agreed 👍 good catch.

> 1. An optional scope MAY be provided after a type. A scope MUST consist of a noun describing a

Reviewer: ```suggestion
1. The scope MAY be provided after a type. A scope MUST consist of a noun describing a
```

  section of the codebase surrounded by parenthesis e.g., `fix(parser):`
1. A description MUST immediately follow the space after the type/scope prefix.
The description is a short summary of the code changes, e.g., _fix: array parsing issue when multiple spaces were contained in string._
1. A longer commit body MAY be provided after the short description, providing additional contextual information about the code changes. The body MUST begin one blank line after the description.
1. A footer MAY be provided one blank line after the body.
  The footer SHOULD contain additional issue references about the code changes (such as the issues it fixes, e.g.,`Fixes #13`).
1. Breaking changes MUST be indicated at the very beginning of the footer or body section of a commit. A breaking change MUST consist of the uppercase text `BREAKING CHANGE`, followed by a colon and a space.
1. A description MUST be provided after the `BREAKING CHANGE: `, describing what has changed about the API, e.g., _BREAKING CHANGE: environment variables now take precedence over config files._

> 1. The footer MUST only contain `BREAKING CHANGE`/description, external links, issue references, and other meta-information.

Reviewer 2: Should it be a comma instead of a slash?

Author: I was trying to indicate that a `BREAKING CHANGE` needs to be terminated by a `: description`. I also feel that this entire line in the spec is pretty redundant with:

> The footer SHOULD contain additional issue references about the code changes (such as the issues it fixes, e.g.,`Fixes #13`).

----

here are a few edits on my mind that perhaps better clarify things:

* Breaking changes MUST be indicated at the very beginning of the body section, or at the beginning of a line in the footer section. A breaking change MUST consist of the uppercase text BREAKING CHANGE, followed by a colon and a space.

*  A footer MAY be provided one blank line after the body.
  The footer MUST contain meta-information about the commit, e.g., related pull-requests, reviewers, breaking changes.

* A footer MAY consist of multiple lines. Each line MUST contain only one piece of meta-information.

* the units of information that make up conventional commits, with the exception of BREAKING CHANGE which MUST be uppercase, MUST NOT be treated as case sensitive by implementors.

Reviewer: ``` `BREAKING CHANGE`/description``` looks strange to me too here

Author: :point_up: @damianopetrungaro what do you think of those four statements above, I was trying to clarify footers a bit: indicating, for instance, that it can be multiple lines and should only have on piece of meta information per line (which is the approach I've seen used in the Chromium and Node.js codebase).

I also thought it's good to add a new rule about capitalization.

Reviewer: I do really like them, especially:
> the units of information that make up conventional commits, with the exception of BREAKING CHANGE which MUST be uppercase, MUST NOT be treated as case sensitive by implementors.

### How should I deal with commit messages in the initial development phase?

> We recommend that you proceed as if you've already released the product. Typically *somebody*, even if its your fellow software developers, is using your software. They'll want to know what's fixed, what breaks etc.

Reviewer 2: `its` => `it's`

* [php-commitizen](https://github.com/damianopetrungaro/php-commitizen): a tool built to create commit messages following the Conventional Commit specs. 
Configurable and usable for PHP projects as a composer dependency or usable globally for non-PHP projects.
* [conform](https://github.com/autonomy/conform): a tool that can be used to enforce policies on git repositories, including conventional commits.

> * [standard-version](https://github.com/conventional-changelog/standard-version): Automatic versioning and CHANGELOG management, using GitHub's new squash button and the recommended Conventional Commits workflow.

Reviewer: Should this be in this PR?

Author: I didn't think it was the end of the world? It was bugging me that `standard-version` was in the wrong section of the footer, and thought I'd bump it up; can pull into another pull request if you prefer.

Reviewer: It's fine :) Let's just create a different commit for that one and I am totally fine with it 😄
